lib/kernel/host/CMakeLists.txt

```diff
@@ -413,7 +407,18 @@ function(compile_sleef VARIANT SLEEF_CONFIG SLEEF_CONFIG_NEW SLEEF_BC)
     file(APPEND "${SLEEF_CONFIG_NEW}" "${STR}")
   endif()
 
-  foreach(VECSIZE "128" "256" "512")
+  # for X86-64, SPIR wrapper is compiled based on largest supported vector size
+  if(HOST_DEVICE_ADDRESS_BITS EQUAL 32)
+    set(SPIR_WRAPPER_FILE "host/spir_wrapper32.ll")
+  else()
+    if(NOT X86)
+      set(SPIR_WRAPPER_FILE "host/spir_wrapper64.ll")
+    else()
+      set(SPIR_WRAPPER_FILE "")
```

Comment:
This could have a comment mentioning the fallback that is applied later in the code?

Reply:
done